Feat/2813 reduce dev env

andrew-jameson opened the pull request:
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2813 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Reduce dev env apps by 1
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?


### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

Commit:
Deletions from global search, need manual intervention in github/circleci cfg

Commit:
Updated diagram

Commit:
removing documentation references to sandbox env

elipe17 reviewed (approved):
Looks good to me! I didn't see any sandbox items alive in cloud.gov and didn't find any other references to it in the repo.

ADPennington committed:
Merge branch 'develop' into feat/2813-reduce-dev-env

ADPennington commented on docs/Technical-Documentation/diagrams/tdp-environments.drawio:
@andrew-jameson did [the PNG version](https://github.com/raft-tech/TANF-app/blob/feat/2813-reduce-dev-env/docs/Technical-Documentation/diagrams/tdp-environments.png) of this diagram get updated too?

ADPennington reviewed (commented):
@andrew-jameson couple of questions on this:

- does the PNG version of system arch diagram also need to be updated?
- I checked the dev environment for apps and services related to sandbox and noticed these services. do these need to be deleted?


```
tdp-storage-sandbox
tdp-redis-sandbox
```

ADPennington reviewed (changes requested):
re: system diagram update. Please also confirm if the additional services I found need to be deleted.

Commit:
Updating png, forgot to export.

andrew-jameson commented:
> @andrew-jameson couple of questions on this:
> 
>     * does the PNG version of system arch diagram also need to be updated?
> 
>     * I checked the dev environment for apps and services related to sandbox and noticed these services. do these need to be deleted?
> 
> 
> ```
> tdp-storage-sandbox
> tdp-redis-sandbox
> ```

Updated png file, deleted redis service. There's an issue with the storage service that I'm working through:

```
"errorResponse": {
    "description": "Service instance tdp-storage-sandbox: Service broker error: BucketNotEmpty: The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
    ....
    {
  "error": {
    "statusCode": 403,
    "status": "403 Forbidden"
  },
  "errorResponse": {
    "description": "You are not authorized to perform the requested action",
    "error_code": "CF-NotAuthorized",
    "code": 10003
  }
}
```

I'll let you know when I've cleared this service out but it won't impact other functionality.

andrew-jameson committed:
Merge branch 'develop' into feat/2813-reduce-dev-env

ADPennington reviewed (approved):
lgtm @andrew-jameson 👍🏾 please check with cloud.gov if we're still blocked from removing sandbox services in dev when this merges. 

test notes [here](https://hackmd.io/P4sQkTRORLGjiodfgRwMtA?view#Feat2813-reduce-dev-env)
